genesyscloud/outbound_callabletimeset/genesyscloud_outbound_callabletimeset_proxy.go

```diff
@@ -0,0 +1,185 @@
+func createOutboundCallabletimesetFn(ctx context.Context, p *outboundCallabletimesetProxy, outboundCallabletimeset *platformclientv2.Callabletimeset) (*platformclientv2.Callabletimeset, error) {
+	callableTimeSet, _, err := p.outboundApi.PostOutboundCallabletimesets(*outboundCallabletimeset)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create Outbound Callabletimeset: %s", err)
+	}
+
+	return callableTimeSet, nil
+}
+
+// getAllOutboundCallabletimesetFn is the implementation for retrieving all Outbound Callabletimeset in Genesys Cloud
+func getAllOutboundCallabletimesetFn(ctx context.Context, p *outboundCallabletimesetProxy) (*[]platformclientv2.Callabletimeset, error) {
+	var allCallabletimesets []platformclientv2.Callabletimeset
+
+	for pageNum := 1; ; pageNum++ {
+		const pageSize = 100
+
+		callableTimeSets, _, err := p.outboundApi.GetOutboundCallabletimesets(pageSize, pageNum, true, "", "", "", "")
+		if err != nil {
+			return nil, fmt.Errorf("Failed to get Outbound Callabletimeset: %v", err)
+		}
+
+		if callableTimeSets.Entities == nil || len(*callableTimeSets.Entities) == 0 {
+			break
+		}
+
+		for _, callableTimeSet := range *callableTimeSets.Entities {
+			log.Printf("Dealing with callableTimeSet id : %s", *callableTimeSet.Id)
+			allCallabletimesets = append(allCallabletimesets, callableTimeSet)
+		}
+	}
+
+	return &allCallabletimesets, nil
+}
+
+// getOutboundCallabletimesetByIdFn is an implementation of the function to get a Genesys Cloud Outbound Callabletimeset by Id
+func getOutboundCallabletimesetByIdFn(ctx context.Context, p *outboundCallabletimesetProxy, id string) (outboundCallabletimeset *platformclientv2.Callabletimeset, statusCode int, err error) {
+	callableTimeSet, resp, err := p.outboundApi.GetOutboundCallabletimeset(id)
+	if err != nil {
+		return nil, resp.StatusCode, fmt.Errorf("Failed to retrieve callabletimeset by id %s: %s", id, err)
+	}
+
+	return callableTimeSet, resp.StatusCode, nil
+}
+
+// getOutboundCallabletimesetIdByNameFn is an implementation of the function to get a Genesys Cloud Outbound Callabletimeset by name
+func getOutboundCallabletimesetIdByNameFn(ctx context.Context, p *outboundCallabletimesetProxy, name string) (id string, retryable bool, err error) {
+	const pageNum = 1
+	const pageSize = 100
+	callableTimeSets, _, err := p.outboundApi.GetOutboundCallabletimesets(pageSize, pageNum, true, "", name, "", "")
+	if err != nil {
+		return "", false, fmt.Errorf("Error searching Outbound Callabletimeset %s: %s", name, err)
+	}
+
+	if callableTimeSets.Entities == nil || len(*callableTimeSets.Entities) == 0 {
+		return "", true, fmt.Errorf("No Outbound Callabletimeset found with name %s", name)
+	}
+
+	if len(*callableTimeSets.Entities) > 1 {
+		return "", false, fmt.Errorf("Too many values returned in look for Outbound Callabletimesets.  Unable to choose 1 Outbound Callabletimeset.  Please refine search and continue.")
+	}
```

Comment:
I would remove this block. It will cause the same behaviour described in [DEVENGAGE-2146](https://inindca.atlassian.net/browse/DEVENGAGE-2146)

Reply:
You just need to loop through the entities and do a exact comparison on the names. Multiple entities could be returned, but only one can match exactly what the set `name` to in the data source. And of course if none match, that's a different error